api/v1alpha1/release_webhook.go

```diff
@@ -17,38 +17,116 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"fmt"
-	"k8s.io/apimachinery/pkg/runtime"
+	"context"
+	"encoding/json"
 	"reflect"
+	"strings"
+
+	"gomodules.xyz/jsonpatch/v2"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
-func (r *Release) SetupWebhookWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
-		Complete()
-}
+const authorLabel = "release.rhtap.openshift.io/author"
 
-//+kubebuilder:webhook:path=/validate-appstudio-redhat-com-v1alpha1-release,mutating=false,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases,verbs=create;update,versions=v1alpha1,name=vrelease.kb.io,admissionReviewVersions=v1
+// releaseHandler describes the handler for the Release low level webhook
+type releaseHandler struct {
```

Comment:
Just call it `webhook` or something like that. It implements more things that only Handler.

Reply:
done in follow up